full_recompile_mcu.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+# Go to the cm_mcu directory (assumes the script is in the cm_mcu directory)
+cd "$(dirname "$0")" || { echo "Failed to enter cm_mcu directory"; exit 1; }
```

Comment:
@prosserlogan suggest replacing all the double-pipe checks with `set -e` once at the start of the script